### PR TITLE
Allows the release namespace to be overridden

### PR DESCRIPTION
**Pull request by KhizerJaan**
**Problem:**
- Vault can only be deployed in Release namespace.

**Root Cause:**
- namespace is hardcoded to `{{ .Release.Namespace }}`

**Solution:**
- Templatize namespaec in _helpers.tpl

### CODE REVIEW

**Comment by tomhjp** [comment]
Hi @KhizerJaan - please could you provide some more context about why this is useful?

**Comment by KhizerJaan** [comment]
If Vault Helm Chart is being used as dependency in another helm (umbrella) chart, it is deployed to umbrella chart release namespace . This PR would enabled to override namespace for Vault Helm Chart.

**Comment by KhizerJaan** [comment]
@tomhjp kindly review and merge the pull request at your convenience. I have umbrella chart use case and am blocked by it . 
By allowing users to choose the deployment namespace, Helm charts become more versatile and adaptable to different environments and deployment strategies. Users can deploy the chart in the namespace that aligns with their specific requirements or organizational practices .

**Comment by tomhjp** [comment]
Thanks for the explanation, that makes sense. This seems like a nice improvement to merge then with a few additions. Please could we add some bats unit tests for each resource, a default empty value in the Values file, an entry in the schema, and a short changelog entry under unreleased improvements? Happy to help out with any of those, just let me know if you need some pointers or assistance.

**Comment by KhizerJaan** [comment]
Sure. Highly appreciate the feedback and suggestions @tomhjp . I will incorporate the suggested changes, and submit a revised version that addresses the feedback provided.

**Comment by KhizerJaan** [comment]
@tomhjp I have incorporated the suggestions provided.
Kindly review the updated PR at your convenience. I believe that it is now complete and addresses all the necessary improvements. However, I welcome any further feedback or suggestions you may have to enhance the quality of the contribution.

**Comment by KhizerJaan** [comment]
@tomhjp  Could you please consider merging this PR? Alternatively, I would greatly appreciate any feedback you might have.

File: values.yaml
Comment (tomhjp) [inline values.yaml:7]:
```suggestion
# The namespace to deploy to. Defaults to the `helm` installation namespace.
```

File: values.yaml
Comment (tomhjp) [inline values.yaml:8]: I think this probably fits best in the `global` section just below.

Comment (tomhjp) [inline values.yaml:8]: WDYT about just `namespace`? It feels descriptive enough to me.

```suggestion
namespace: ""
```

File: CHANGELOG.md
Comment (tomhjp) [inline CHANGELOG.md:51]:
```suggestion
* global: Add `global.namespace` to override the helm installation namespace. [GH-909](https://github.com/hashicorp/vault-helm/pull/909)
```

File: test/unit/csi-agent-configmap.bats
Comment (tomhjp) [inline test/unit/csi-agent-configmap.bats:24]: Thanks for adding these tests, they look great.

**Review by tomhjp — COMMENTED** [review]
Thanks for the changes so far - just a few more small suggestions and then I think this is ready to merge.

**Comment by tomhjp** [comment]
I ran the CI and a few unit test failures came up FYI:

```text
not ok 166 injector/certs-secret: namespace is set in 131ms
not ok 168 injector/role: namespace is set in 130ms
not ok 170 injector/rolebinding: namespace is set in 135ms
not ok 174 injector/MutatingWebhookConfiguration: namespace is set in 138ms
```

**Comment by KhizerJaan** [comment]
@tomhjp Thank you for suggestions . highly appreciate it . I've updated regards to your suggestions . PTAL

**Comment by KhizerJaan** [comment]
@tomhjp PTAL

**Comment by KhizerJaan** [comment]
@tomhjp can this be merged please.

File: values.yaml
Comment (tomhjp) [inline values.yaml:6]:
```suggestion
```

**Review by tomhjp — APPROVED** [review]
Sorry, the changelog changed in main since my last review - please could you rebase/merge in main to make sure the changes stay in the unreleased section when this is merged? Otherwise LGTM

**Comment by KhizerJaan** [comment]
@tomhjp I can't merge this . can you please merge.

**Comment by KhizerJaan** [comment]
@tomhjp can this be merged please .

**Comment by KhizerJaan** [comment]
@tomhjp  branch synced . PTAL

**Comment by KhizerJaan** [comment]
@tomhjp can you please merge this PR? I've synced with main . or it would be great if yo have any feedback.

**Comment by tomhjp** [comment]
I've just run the tests again to prepare for merging, but it looks like the same ones from before are still failing. Do you need some help getting going with running them locally?

```
not ok 166 injector/certs-secret: namespace is set
not ok 168 injector/role: namespace is set
not ok 170 injector/rolebinding: namespace is set
not ok 174 injector/MutatingWebhookConfiguration: namespace is set
```

**Comment by KhizerJaan** [comment]
@tomhjp  previously results were escaped like this `\"bar\"` and it was failing locally so i removed escaping characters . now I've reverted back to escaping . can you please test now?

**Comment by tomhjp** [comment]
Looks like that's done the trick, thanks. The other way to fix it would be to add `-r` to the `yq` command, like `yq -r '.webhooks[0].clientConfig.service.namespace'`, but what you've got now LGTM.

**Pull request merged**
